kasa/interfaces/light.py

```diff
@@ -196,3 +172,42 @@ def state(self) -> LightState:
     @abstractmethod
     async def set_state(self, state: LightState) -> dict:
         """Set the light state."""
+
+    def _deprecated_valid_temperature_range(self) -> ColorTempRange:
+        if not (temp := self.get_feature("color_temp")):
+            raise KasaException("Color temperature not supported")
+        return ColorTempRange(temp.minimum_value, temp.maximum_value)
+
+    def _deprecated_attributes(self, dep_name: str) -> Callable | None:
+        map: dict[str, Callable] = {
+            "is_color": self.set_hsv,
+            "is_dimmable": self.set_brightness,
+            "is_variable_color_temp": self.set_color_temp,
+        }
+        return map.get(dep_name)
+
+    def __getattr__(self, name: str) -> Any:
+        if name == "valid_temperature_range":
+            res = self._deprecated_valid_temperature_range()
+            msg = (
+                "valid_temperature_range is deprecated, use "
+                "get_feature(self.set_color_temp) minimum_value "
+                " and maximum_value instead"
+            )
+            warn(msg, DeprecationWarning, stacklevel=2)
+            return res
+
+        if name == "has_effects":
+            msg = (
+                "has_effects is deprecated, use Module.LightEffect "
+                "in device.modules instead"
```

Comment:
```suggestion
                "has_effects is deprecated, check `Module.LightEffect "
                "in device.modules` instead"
```
Or something like that.

Reply:
Done